packages/core/src/types.ts

```diff
@@ -206,7 +214,7 @@ export type WithDynamicParams<
       {
         type: T['type'];
         params?:
-          | T['params']
+          | NotAFunction<T['params']>
```

Comment:
if we decide to land this PR then this helper should be added to many more places